tests/initialize_holder_rewards.rs

```diff
@@ -0,0 +1,459 @@
+                .unwrap(),
+        );
+    }
+
+    let instruction =
+        initialize_holder_rewards(&holder_rewards_pool, &holder_rewards, &token_account, &mint);
+
+    let transaction = Transaction::new_signed_with_payer(
+        &[instruction],
+        Some(&context.payer.pubkey()),
+        &[&context.payer],
+        context.last_blockhash,
+    );
+
+    let err = context
+        .banks_client
+        .process_transaction(transaction)
+        .await
+        .unwrap_err()
+        .unwrap();
+
+    assert_eq!(
+        err,
+        TransactionError::InstructionError(0, InstructionError::InvalidAccountOwner)
+    );
+}
+
+#[tokio::test]
+async fn fail_holder_rewards_incorrect_address() {
+    let owner = Pubkey::new_unique();
+    let mint = Pubkey::new_unique();
+
+    let token_account = get_associated_token_address(&owner, &mint);
+    let holder_rewards = Pubkey::new_unique(); // Incorrect holder reward address.
+    let holder_rewards_pool = get_holder_rewards_pool_address(&mint);
+
+    let mut context = setup().start_with_context().await;
+    setup_holder_rewards_pool_account(&mut context, &holder_rewards_pool, 0).await;
+    setup_token_account(&mut context, &token_account, &owner, &mint, 0).await;
+    setup_mint(&mut context, &mint, &Pubkey::new_unique(), 0).await;
+
+    let instruction =
+        initialize_holder_rewards(&holder_rewards_pool, &holder_rewards, &token_account, &mint);
+
+    let transaction = Transaction::new_signed_with_payer(
+        &[instruction],
+        Some(&context.payer.pubkey()),
+        &[&context.payer],
+        context.last_blockhash,
+    );
+
+    let err = context
+        .banks_client
+        .process_transaction(transaction)
+        .await
+        .unwrap_err()
+        .unwrap();
+
+    assert_eq!(
+        err,
+        TransactionError::InstructionError(
+            0,
+            InstructionError::Custom(PaladinRewardsError::IncorrectHolderRewardsAddress as u32)
+        )
+    );
+}
+
+#[tokio::test]
+async fn fail_holder_rewards_account_initialized() {
+    let owner = Pubkey::new_unique();
+    let mint = Pubkey::new_unique();
+
+    let token_account = get_associated_token_address(&owner, &mint);
+    let holder_rewards = get_holder_rewards_address(&token_account);
+    let holder_rewards_pool = get_holder_rewards_pool_address(&mint);
+
+    let mut context = setup().start_with_context().await;
+    setup_holder_rewards_pool_account(&mut context, &holder_rewards_pool, 0).await;
+    setup_token_account(&mut context, &token_account, &owner, &mint, 0).await;
+    setup_mint(&mut context, &mint, &Pubkey::new_unique(), 0).await;
+
+    // Set up an already (arbitrarily) initialized holder rewards account.
+    {
+        context.set_account(
+            &holder_rewards,
+            &AccountSharedData::from(Account {
+                lamports: 1_000_000_000,
+                data: vec![2; 45],
```

Comment:
Can this be the same size as a real holder rewards account? Although it'll trip up the same error, it's a more interesting case, of someone trying to reinitialize the holder rewards

Reply:
Done!